aip/auth/4122.md

```diff
@@ -0,0 +1,349 @@
+---
+id: 4122
+scope: auth
+state: reviewing
+created: 2020-12-15
+---
+
+# Service Account Impersonation
+
+A service account or a user identity in GCP can impersonate another service 
+account, and request short-lived credentials on behalf of that service account.
+
+The request can be direct or delegated.
+
+In the direct case, the caller is authenticated as either a Google account or a
+service account, and makes a direct request to create short-lived credentials on
+behalf of another service account. The requesting identity must have the Service
+Account Token Creator role (`roles/iam.serviceAccountTokenCreator`) on the target
+service account.
+
+In the delegated case, the caller is also authenticated as either a Google
+account or a service account, but the request goes through one or more service
+accounts in a delegation chain. Each identity in the chain must grant the Service
+Account Token Creator role to the previous identity in the chain. 
+
+**Note:** Because this AIP describes guidance and requirements in a
+language-neutral way, it uses generic terminology which may be imprecise or
+inappropriate in certain languages or environments.
+
+## Guidance
+
+This section describes the general guidance for supporting Service Account
```

Comment:
I'd really like to see a typical flow of requests and responses without any reference to auth libraries. It's much easier to understand the library behavior when we've got a clearer idea of what it's trying to implement.